File: 02-java-security-features/mem-safety/Memory.java
public class Memory {
    public static void main(String[] args) {
        int[] buf = new int[32];
            if (args.length != 1) {
            System.err.println("Provide one numeric argument.");
            System.exit(-1);
        }
        try {
            int index = Integer.parseInt(args[0]);
            System.out.println("buf[" + index + "] is " + buf[index]);
            buf[index] = 1000;
            //System.out.println("buf[40] is " + buf[40]);
            //buf[40] = 1000;
        } catch (NumberFormatException e) {
            System.err.println("Provide one _numeric_ argument.");
        }
    }
}

File: 02-java-security-features/mem-safety/memory.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
	unsigned int buf[32];
	long index;
	char *endptr;

	if (argc != 2) {
		fprintf(stderr, "Usage: %s <index>\n", argv[0]);
		exit(EXIT_FAILURE);
	}

	index = strtol(argv[1], &endptr, 10);
	if (*endptr != '\0') {
		fprintf(stderr, "Use integer as argument\n");
		exit(EXIT_FAILURE);
	}

	printf("buf[%ld] is %u\n", index, buf[index]);
	buf[index] = 1000;

	return 0;
}
